helper/resource/testing_new_import_state.go

```diff
@@ -135,7 +135,14 @@ func testStepNewImportState(t *testing.T, c TestCase, wd *tftest.WorkingDir, ste
 			// this shouldn't happen in any reasonable case.
 			// KEM CHANGE FROM OLD FRAMEWORK: Fail test if this happens.
 			var rsrcSchema *schema.Resource
-			providerAddr, diags := addrs.ParseAbsProviderConfigStr("provider." + r.Provider + "." + r.Type)
+
+			// r.Provider at this point is `registry.terraform.io/hashicorp/blah` but we need `blah`
+			segments := strings.Split(r.Provider, "/")
```

Comment:
I'd prefer if we could use the core code for parsing this string instead of relying on `strings.Split`; I think it makes it a little more robust and offers a little more flexibility about input, accepting the old version _and_ the new version at the same time?

Reply:
sure, can you point me towards this?

Reply:
https://pkg.go.dev/github.com/hashicorp/terraform/addrs#ParseProviderSourceString is what I had in mind

Reply:
done

Reply:
I'd rather not import the whole of Core into the SDK to solve this small problem, after we spent so much effort decoupling it (for reasons I won't repeat here). The `strings.Split` solution is fine for now, or you can use https://github.com/hashicorp/terraform-registry-address.

Reply:
Can't use that package, it's not public yet.

Once https://github.com/hashicorp/terraform-plugin-test/pull/44 is merged I've got a local fix for this.

Reply:
What kind of local fix - do you mean a way to avoid importing Core?

Reply:
Yeah, I copied over the functions instead of importing them.